libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h

```diff
@@ -150,6 +150,8 @@ typedef struct AzureIoTADUClientDeviceProperties
     uint32_t ulModelLength;
 
     AzureIoTADUDeviceCustomProperties_t * pxCustomProperties;
+    const uint8_t * ucCompatibilityProperties;
+    uint32_t ulCompatibilityPropertiesLength;
```

Comment:
should these live here? Since they're technically part of separate fields? 

![image](https://user-images.githubusercontent.com/36710865/179790660-3279fc90-0ded-4476-91e5-e204ff40e29e.png)

Maybe have it as the first candidate for a field in our `options` struct? It being in the options also allows us to set it to the default for the case that the user doesn't set it?

Reply:
Done.